src/main/java/org/bigredbands/mb/controllers/MainController.java

```diff
@@ -345,32 +334,35 @@ public void changeMoves(int targetMove) {
         if (targetMove != currentMove) {
             currentMove = targetMove;
             if (!selectedRanks.isEmpty()) {
-                mainView.updateSelectedRank(selectedRanks, getSharedCommands(selectedRanks, drillInfo.getMoves().get(currentMove).getCommands()));
+                mainView.updateSelectedRank(selectedRanks);
             }
             mainView.updateView(currentMove, drillInfo.getMoves().get(currentMove).getCounts());
         }
     }
 
     /**
-     * Adds a new rank to the models and update the view
+     * Adds a new rank to the models and update the view.
      *
-     * @param name - the name of the new rank
-     * @param rankPosition - the position of the new rank
+     * @param name - The name of the new rank
+     * @param rankPosition - The position of the new rank
+     * @return - An error messsge iff adding a rank failed. An empty string
+     *         otherwise.
      */
     @Override
-    public void addRank(String rankName, RankPosition rankPosition) {
-        String errorMessage = drillInfo.addRankToMoves(rankName, rankPosition);
-        if (errorMessage.isEmpty()) {
-            // TODO: maybe not clear? idk
-            selectedRanks.clear();
-            selectedRanks.add(rankName);
-            modified = true;
-            mainView.updateSelectedRank(selectedRanks, getSharedCommands(selectedRanks,drillInfo.getMoves().get(currentMove).getCommands()));
-            mainView.updateView(currentMove, drillInfo.getMoves().get(currentMove).getCounts());
-        }
-        else {
-            mainView.displayError(errorMessage);
+    public String addRank(String rankName, RankPosition rankPosition) {
+        final String errorMessage = drillInfo.addRankToMoves(rankName, rankPosition);
+        if (!errorMessage.isEmpty()) {
+            return errorMessage;
+            
         }
```

Comment:
Eventually I feel like errors should be propagated via exceptions (rather than string return values), but since we're already using the return value for them in `drillInfo` I don't mind it much.